dcos/packagemanager.py

```diff
@@ -245,27 +245,42 @@ def remove_repo(self, name):
         response = self.cosmos_post("repository/delete", params=params)
         return response.json()
 
-    def package_add(self, dcos_package):
+    def package_add(self, dcos_package, package_name, package_version):
         """
-        Adds a DC/OS package to DC/OS
+         Adds a DC/OS package to DC/OS
 
         :param dcos_package: path to the DC/OS package
-        :type dcos_package: str
+        :type dcos_package: None | str
+        :param package_name: name of the remote pacakge to add
+        :type package_name: None | str
+        :param package_version: version of the remote package to add
+        :type package_version: None | str
         :return: Response to the package add request
         :rtype: Response
         """
-        with util.open_file(dcos_package, 'rb') as pkg:
-            extra_headers = {'Content-MD5': util.md5_hash_file(pkg)}
-        with util.open_file(dcos_package, 'rb') as data:
-            try:
-                return self._post('add', headers=extra_headers, data=data)
-            except DCOSHTTPException as e:
-                if e.status() == 404:
-                    message = 'Your version of DC/OS ' \
-                              'does not support this operation'
-                    raise DCOSException(message)
-                else:
-                    raise e
+        try:
+            if dcos_package:
+                with util.open_file(dcos_package, 'rb') as pkg:
+                    extra_headers = {
+                        'Content-Type':
+                            'application/vnd.dcos.'
+                            'universe.package+zip;version=v1',
+                        'Content-MD5': util.md5_hash_file(pkg)
+                    }
+                with util.open_file(dcos_package, 'rb') as data:
+                    return self._post('add', headers=extra_headers, data=data)
+            else:
+                json = {'packageName': package_name}
+                if package_version is not None:
+                    json['packageVersion'] = package_version
+                return self._post('add', params=json)
+        except DCOSHTTPException as e:
```

Comment:
this is too broad of an except. How are we determining this new feature. can we add a cosmos capability?